src/components/data-exploration/heatmap/HeatmapMetadataTrackSettings.jsx

```diff
@@ -19,32 +19,32 @@ const HeatmapMetadataTrackSettings = (props) => {
 
   const { componentType } = props;
   const selectedHierarchy = useSelector(getCellSetsHierarchy(['cellSets', 'metadataCategorical']));
-  const currentSelectedHierarchy = useRef(selectedHierarchy);
+
   const selectedTracks = useSelector(
     (state) => state.componentConfig[componentType].config.selectedTracks,
   );
+
   const getTrackData = () => selectedHierarchy.map(
     (data) => ({ selected: selectedTracks.includes(data.key), key: data.key }),
   );
 
   const isInitialRenderRef = useRef(true);
+
   const [trackData, setTrackData] = useState(getTrackData());
 
   const getUpdatedTrackData = () => _.unionBy(
     getTrackData(),
     trackData,
     'key',
   );
+
   useEffect(() => {
     // Prevent initial dispatch when object appears
-    if (isInitialRenderRef.current
-      || _.isEqual(selectedHierarchy, currentSelectedHierarchy.current)) {
-      return;
-    }
-    currentSelectedHierarchy.current = selectedHierarchy;
-    // Do not re-render if visible track data hasn't changed
```

Comment:
This had been added due to `getCellSetsHierarchy` generating an array with the same values but different ref, so `useSelector` wouldn't recognize that it hadn't changed and would rerun.

Now it is no longer necessary

Reply:
it's great that we can remove this